specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+Action Centre. We can perform a number of operations on these notifications such as retrieval and
+deletion.
+
+```cpp
+// Get a List of all active notifications displayed in the Action Centre
+winrt::Windows::Foundation::IAsyncOperation<Windows::Foundation::Collections::IVector<AppNotification>> GetNotificationListASync()
+{
+    auto notifications = co_await AppNotificationManager::Default().GetAllAsync();
+    co_return notifications;
+}
+
+// Remove a notification from action centre using tag and group
+winrt::Windows::Foundation::IAsyncAction RemoveNotification(const winrt::hstring tag, const winrt::hstring group)
+{
+    co_await AppNotificationManager::Default().RemoveWithTagGroupAsync(tag, group);
+}
+```
+
+## Notification Progress Updates
+
+Sometimes a developer would like to show progress bar related updates in a notification:
+
+![Notification Progress](https://docs.microsoft.com/windows/apps/design/shell/tiles-and-notifications/images/toast-progressbar-annotated.png)
+
+To accomplish that, the developer will need to use the AppNotificationProgressData construct to update the progress bar on the screen.
+
+```cpp
+// Send first Notification Progress Update
+void SendUpdatableNotificationWithProgress()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    winrt::hstring payload =
+    LR"(<toast launch="action = viewDownload &amp; downloadId = 9438108">
+        <visual>
+            <binding template = "ToastGeneric">
+                <text>Downloading this week's new music...</text>
+                <progress
+                    title = "{progressTitle}
+                    value = "{progressValue}"
+                    valueStringOverride = "{progressValueString}"
+                    status = "{progressStatus}" />
+            </binding>
+        </visual>
+    </toast>)";
+
+    XmlDocument doc;
+    doc.LoadXml(payload);
+    AppNotification notification(doc);
+    notification.Tag(tag);
+    notification.Group(group);
+
+    // Assign initial values for first notification progress UI
+    AppNotificationProgressData data;
+    data.Title(L"Weekly playlist"); // Binds to {progressTitle} in xml payload
+    data.Value(0.6); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"15/26 songs"); // Binds to {progressValueString} in xml payload
+    data.Status(L"Downloading..."); // Binds to {progressStatus} in xml payload
+    data.SequenceNumber(1);
+
+    notification.Progress(data);
+    AppNotificationManager::Default().Show(notification);
+}
+
+// Send subsequent progress updates
+winrt::Windows::Foundation::IAsyncAction UpdateProgressAsync()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    // Assign new values
+    // Note that you only need to assign values that changed. In this example we don't assign progressStatus since we don't need to change it
```

Comment:
DISCUSSION: The comment about "You only need to assign values that changed" may be incorrect. Please investigate and fix.

if it's correct, then we have open questions about how to detect which values have changed.

Reply:
The developer is expected to refresh each and every property in ProgressData. Otherwise, we assume that the property default (empty string in this case) takes over.